src/main/kotlin/org/lfenergy/gxf/protocol/adapter/oslp/mikronika/device/communication/signing/SigningConfiguration.kt

```diff
@@ -3,14 +3,17 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.lfenergy.gxf.protocol.adapter.oslp.mikronika.device.communication.signing
 
-import org.springframework.boot.context.properties.ConfigurationProperties
+import com.gxf.utilities.oslp.message.signing.SigningUtil
+import com.gxf.utilities.oslp.message.signing.configuration.SigningProperties
+import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 
 @Configuration
-@ConfigurationProperties(prefix = "signing")
 class SigningConfiguration {
-    lateinit var securityProvider: String
-    lateinit var securityAlgorithm: String
-    lateinit var securityKeyType: String
-    lateinit var privateKeyPath: String
+    @Bean
+    fun signingUtil(signingConfiguration: SigningConfigurationProperties): SigningUtil {
```

Comment:
Consider renaming the parameter, as it now is sort of the same as the name of this class.